docs/specs/captcha.md

```diff
@@ -1,41 +1,460 @@
-# Captcha
+# Risk assessment and Captcha
 
-- [Configurations](#configurations)
-  - [authgear.yaml](#authgearyaml)
-  - [authgear.secrets.yaml](#authgearsecretsyaml)
-- [Supported Workflows](#supported-workflows)
+- [Risk assessment and Captcha](#risk-assessment-and-captcha)
+  * [Old configuration](#old-configuration)
+  * [New Configuration](#new-configuration)
+    + [authgear.yaml](#authgearyaml)
+    + [authgear.secrets.yaml](#authgearsecretsyaml)
+  * [Authentication Flow](#authentication-flow)
+    + [Risk assessment and Captcha in Authentication Flow configuration](#risk-assessment-and-captcha-in-authentication-flow-configuration)
+    + [Behavior of generated flows](#behavior-of-generated-flows)
+    + [Risk assessment and Captcha in Authentication Flow API](#risk-assessment-and-captcha-in-authentication-flow-api)
+    + [Advanced use case: Require Captcha at a specific branch only](#advanced-use-case-require-captcha-at-a-specific-branch-only)
+    + [Advanced use case: Use different Captcha providers in different branches](#advanced-use-case-use-different-captcha-providers-in-different-branches)
+    + [Advanced use case: Use fail-open instead of fail-close](#advanced-use-case-use-fail-open-instead-of-fail-close)
+    + [Advanced use case: Allow internal staff to bypass Captcha](#advanced-use-case-allow-internal-staff-to-bypass-captcha)
+    + [Advanced use case: Require Captcha only when risk level is high](#advanced-use-case-require-captcha-only-when-risk-level-is-high)
+  * [Audit log](#audit-log)
+  * [Study on Captcha providers](#study-on-captcha-providers)
+    + [Geetest v4](#geetest-v4)
+    + [Geetest v3](#geetest-v3)
+    + [hCaptcha](#hcaptcha)
+    + [reCAPTCHA v2](#recaptcha-v2)
+    + [reCAPTCHA v3](#recaptcha-v3)
+    + [reCAPTCHA Enterprise](#recaptcha-enterprise)
+    + [Cloudflare Turnstile](#cloudflare-turnstile)
+    + [Arkose Labs Bot Manager](#arkose-labs-bot-manager)
+    + [Tencent Captcha](#tencent-captcha)
 
-## Configurations
+## Old configuration
 
-### authgear.yaml
+> The old configuration **IS NOT** used by Authentication Flow.
+> To configure Captcha providers for an Authentication Flow, the new configuration must be used.
+
+See [here](./captcha_legacy.md#configuration) for the documentation of the old configuration.
+
+## New Configuration
+
+The section documents the new configuration.
 
-Before using any workflow with captcha enabled, the captcha provider must be configured in `authgear.yaml`:
+### authgear.yaml
 
 ```yaml
+risk_assessment:
+  enabled: true
+  providers:
+  - type: recaptchav3
+    alias: recaptchav3
+    site_key: "SITE_KEY"
+    risk_level:
+      high_if_gte: 0.7
+      medium_if_gte: 0.5
+
 captcha:
-  provider: cloudflare
+  enabled: true
+  ip_allowlist:
```

Comment:
does this also applies to `risk_assessment` automatically?

Reply:
Yes, it should.

Reply:
maybe specify it or put the risk section under captcha as well.

Reply:
OK.